src/python/pants/option/config.py

```diff
@@ -69,40 +68,40 @@ def load(
         DEFAULT section, and be available for use in substitutions.  The caller may override some of
         these seed values.
         """
-        single_file_configs = []
+        config_values = []
         for file_content in file_contents:
-            content_digest = sha1(file_content.content).hexdigest()
             normalized_seed_values = cls._determine_seed_values(seed_values=seed_values)
-
             try:
-                config_values = cls._parse_toml(
-                    file_content.content.decode(), normalized_seed_values
-                )
+                config_values.append(cls._parse_toml(file_content, normalized_seed_values))
             except Exception as e:
                 raise ConfigError(
                     f"Config file {file_content.path} could not be parsed as TOML:\n  {e}"
                 )
-
-            single_file_configs.append(
-                _SingleFileConfig(
-                    config_path=file_content.path,
-                    content_digest=content_digest,
-                    values=config_values,
-                ),
-            )
-        return _ChainedConfig(tuple(reversed(single_file_configs)))
+        return cls(tuple(config_values))
 
     @classmethod
     def _parse_toml(
-        cls, config_content: str, normalized_seed_values: dict[str, str]
+        cls, config_source: ConfigSource, normalized_seed_values: dict[str, str]
     ) -> _ConfigValues:
         """Attempt to parse as TOML, raising an exception on failure."""
-        toml_values = cast(Dict[str, Any], toml.loads(config_content))
-        toml_values["DEFAULT"] = {
+        toml_values = cast(Dict[str, Any], toml.loads(config_source.content.decode()))
+        toml_values[DEFAULT_SECTION] = {
             **normalized_seed_values,
-            **toml_values.get("DEFAULT", {}),
+            **toml_values.get(DEFAULT_SECTION, {}),
         }
-        return _ConfigValues(toml_values)
+        return _ConfigValues(config_source.path, toml_values)
+
+    def verify(self, section_to_valid_options: dict[str, set[str]]):
```

Comment:
Having the verification logic in Options.verify_configs() was clunky as it required poking around inside config classes. Moving it here streamlines things, but requires passing in a dict of the valid options (as known by Options.verify_configs()).